internal/dbutil/sqliteutil/migrator.go

```diff
@@ -20,57 +28,60 @@ func MigrateTablesToDatabase(ctx context.Context, srcDB, destDB *sql.DB, tablesT
 	// we can execute the backup API for a corruption safe clone.
 	srcConn, err := srcDB.Conn(ctx)
 	if err != nil {
-		return errs.Wrap(err)
+		return ErrMigrateTables.Wrap(err)
 	}
 
+	defer func() {
+		err = errs.Combine(err, ErrMigrateTables.Wrap(srcConn.Close()))
+	}()
+
 	destConn, err := destDB.Conn(ctx)
 	if err != nil {
-		return errs.Wrap(err)
+		return ErrMigrateTables.Wrap(err)
 	}
 
+	defer func() {
+		err = errs.Combine(err, ErrMigrateTables.Wrap(destConn.Close()))
+	}()
+
 	// The references to the driver connections are only guaranteed to be valid
 	// for the life of the callback so we must do the work within both callbacks.
 	err = srcConn.Raw(func(srcDriverConn interface{}) error {
 		srcSqliteConn, ok := srcDriverConn.(*sqlite3.SQLiteConn)
 		if !ok {
-			return errs.New("unable to get database driver")
+			return ErrMigrateTables.New("unable to get database driver")
 		}
 
-		err = destConn.Raw(func(destDriverConn interface{}) error {
+		err := destConn.Raw(func(destDriverConn interface{}) error {
 			destSqliteConn, ok := destDriverConn.(*sqlite3.SQLiteConn)
 			if !ok {
-				return errs.New("unable to get database driver")
+				return ErrMigrateTables.New("unable to get database driver")
 			}
 
-			err = backup(ctx, srcSqliteConn, destSqliteConn)
+			err := backup(ctx, srcSqliteConn, destSqliteConn)
 			if err != nil {
-				return errs.New("unable to backup database")
+				return ErrMigrateTables.Wrap(err)
 			}
 			return nil
 		})
 		if err != nil {
-			return errs.Wrap(err)
+			return ErrMigrateTables.Wrap(err)
 		}
 
 		return nil
 	})
 	if err != nil {
-		return errs.Wrap(err)
+		return ErrMigrateTables.Wrap(err)
 	}
 
-	if err := srcConn.Close(); err != nil {
-		return errs.Wrap(err)
-	}
-	if err := destConn.Close(); err != nil {
-		return errs.Wrap(err)
+	err = errs.Combine(err, srcConn.Close(), destConn.Close())
```

Comment:
This would end-up as a double close, which in some cases could cause an error or panic in some cases. When things add wrappings etc. handling, they expect usually that `Close` is only ever called once. As an example double `close` on channels panics.

Reply:
I see that now. We could add a bool `connsClosed` that is checked in the defer func to avoid the double close.

Reply:
I see what you mean now by another function. I'd probably prefer to do it that way, I'll push up another commit so you can see what I'm thinking.

Reply:
Yeah, but then this would end up more complicated :D. You can definitely try how it would look.

With closure it's clear, that "when this func ends, this defer will be called". However, it does look quite ugly.

The alternative approach to closure could be to split this func into two. Add `Backup` that handles that does the backing up things, then you won't have the ugly closure, however you would have an extra func... which in this case might even end up clearer.

Usually with closing things I do:

1. add creation
2. defer close (or if it's attached to it, then add it to the appropriate close)

This mostly ensures that I don't accidentally forget or miss a close somewhere.

Reply:
Yeah I was suggesting in my second comment that I would probably prefer to split into two functions. I'll push up a prototype so we can see. I'm pairing right now, but soon I'll do that.